### PR TITLE
E2Es based on EMB case studies

### PR DESCRIPTION
**Pull request by seran**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -29,7 +29,7 @@
         <module>test-old-libraries</module>
         <module>solver</module>
         <module>test-utils</module>
-        <module>wfc</module> <!-- TODO remove once moved in own repository-->
+        <module>wfc</module>
```
Comment (arcuri82) [inline pom.xml:32]: @seran why removing the comment?

Comment (seran) [reply pom.xml:32]: One of the previous commit has this comment next to `<module>e2e-tests/emb-json</module>`. Seems like confusion created by IntelliJ when I was creating the module. When I removed the new module from the `pom.xml` this comment also got removed. I'll fix it.

File: e2e-tests/emb-json/target/classes/META-INF/emb-json.kotlin_module
[the remark this answers is not in the document]
Comment (seran) [reply e2e-tests/emb-json/target/classes/META-INF/emb-json.kotlin_module:1]: I vividly remember removing this file yesterday. Seems like this got added from another commit `cc82d32cf7ebf549e0390b25fbfac36b225510b6` from a different PR into `master`.

File: e2e-tests/emb-json/src/main/java/com/foo/rest/emb/json/language/Tools.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import java.net.MalformedURLException;
+import java.net.URL;
+
+/**
+ * This code is taken from LanguageTool
+ * G: https://github.com/languagetool-org/languagetool
+ * L: LGPL-2.1
+ * P: src/main/java/org/languagetool/tools/Tools.java
+ */
+public class Tools {
```
Comment (arcuri82) [inline e2e-tests/emb-json/src/main/java/com/foo/rest/emb/json/language/Tools.java:12]: is this used anywhere?

File: e2e-tests/emb-json/src/main/java/com/foo/rest/emb/json/ocvn/ReleaseJsonToObject.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    public ReleaseJsonToObject(final String jsonObject, ObjectMapper mapper) {
+        this.mapper = mapper;
+        this.jsonObject = jsonObject;
+    }
+
+
+    public Release toObject() throws IOException {
+        if (release == null) {
+            // Transform JSON String to a Release Object
+            release = this.mapper.readValue(this.jsonObject, Release.class);
```
Comment (arcuri82) [inline e2e-tests/emb-json/src/main/java/com/foo/rest/emb/json/ocvn/ReleaseJsonToObject.java:28]: didn't we already handled this case? if so, i guess we could remove OCVN from here

File: e2e-tests/emb-json/src/main/resources/countriesV2.json
Comment (arcuri82) [inline e2e-tests/emb-json/src/main/resources/countriesV2.json:1]: is this file still needed?

File: e2e-tests/emb-json/src/test/java/org/evomaster/e2etests/emb/json/gestaohospital/GestaoHospitalExampleEMTest.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    @BeforeAll
+    public static void initClass() throws Exception {
+        GestaoHospitalController controller = new GestaoHospitalController();
+        EMConfig config = new EMConfig();
+        config.getInstrumentMR_EXT_0();
+        EMBJsonTestBase.initClass(controller, config);
+    }
+
+    @Disabled
+    public void runEMTest() throws Throwable {
```
Comment (arcuri82) [inline e2e-tests/emb-json/src/test/java/org/evomaster/e2etests/emb/json/gestaohospital/GestaoHospitalExampleEMTest.java:23]: when having a disabled test, still must have the annotation `@Test`

File: e2e-tests/emb-json/src/test/java/org/evomaster/e2etests/emb/json/language/LanguageServerExampleEMTest.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    @BeforeAll
+    public static void initClass() throws Exception {
+        LanguageServerExampleController controller = new LanguageServerExampleController();
+        EMConfig config = new EMConfig();
+        config.getInstrumentMR_EXT_0();
+        EMBJsonTestBase.initClass(controller, config);
+    }
+
+    @Disabled
+    public void runEMTest() throws Throwable {
```
Comment (arcuri82) [inline e2e-tests/emb-json/src/test/java/org/evomaster/e2etests/emb/json/language/LanguageServerExampleEMTest.java:23]: when having a disabled test, still must have the annotation `@Test`

File: e2e-tests/emb-json/src/test/java/org/evomaster/e2etests/emb/json/proxyprint/ProxyPrintExampleEMTest.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    @BeforeAll
+    public static void initClass() throws Exception {
+        ProxyPrintExampleController controller = new ProxyPrintExampleController();
+        EMConfig config = new EMConfig();
+        config.getInstrumentMR_EXT_0();
+        EMBJsonTestBase.initClass(controller, config);
+    }
+
+    @Disabled
+    public void runEMTest() throws Throwable {
```
Comment (arcuri82) [inline e2e-tests/emb-json/src/test/java/org/evomaster/e2etests/emb/json/proxyprint/ProxyPrintExampleEMTest.java:23]: when having a disabled test, still must have the annotation `@Test`

File: e2e-tests/emb-json/src/test/java/org/evomaster/e2etests/emb/json/tiltaksgjennomforing/TiltaksgjennomforingExampleEMTest.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    @BeforeAll
+    public static void initClass() throws Exception {
+        TiltaksgjennomforingExampleController controller = new TiltaksgjennomforingExampleController();
+        EMConfig config = new EMConfig();
+        config.getInstrumentMR_EXT_0();
+        EMBJsonTestBase.initClass(controller, config);
+    }
+
+    @Disabled
+    public void runEMTest() throws Throwable {
```
Comment (arcuri82) [inline e2e-tests/emb-json/src/test/java/org/evomaster/e2etests/emb/json/tiltaksgjennomforing/TiltaksgjennomforingExampleEMTest.java:23]: when having a disabled test, still must have the annotation `@Test`

File: e2e-tests/emb-json/src/test/java/org/evomaster/e2etests/emb/json/gestaohospital/GestaoHospitalExampleEMTest.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+        // Code reach to a point to detect it should be an array.
+        // ["_EM_11_XYZ_","_EM_11_XYZ_","_EM_11_XYZ_"]
+        // But for this, the array should contain a custom object as elements
+        // where it fails.
+        runTestHandlingFlakyAndCompilation(
```
Comment (arcuri82) [inline e2e-tests/emb-json/src/test/java/org/evomaster/e2etests/emb/json/gestaohospital/GestaoHospitalExampleEMTest.java:28]: i can look into it, but in a different branch

Comment (arcuri82) [reply e2e-tests/emb-json/src/test/java/org/evomaster/e2etests/emb/json/gestaohospital/GestaoHospitalExampleEMTest.java:28]: same for the other disabled tests

**Review by arcuri82 — CHANGES_REQUESTED** [review]

File: e2e-tests/emb-json/src/test/java/org/evomaster/e2etests/emb/json/gestaohospital/GestaoHospitalExampleEMTest.java (hunk 8 above)
Comment (seran) [reply e2e-tests/emb-json/src/test/java/org/evomaster/e2etests/emb/json/gestaohospital/GestaoHospitalExampleEMTest.java:28]: 👍

File: e2e-tests/emb-json/src/main/java/com/foo/rest/emb/json/ocvn/ReleaseJsonToObject.java (hunk 3 above)
Comment (seran) [reply e2e-tests/emb-json/src/main/java/com/foo/rest/emb/json/ocvn/ReleaseJsonToObject.java:28]: For now, this is the only case study which is working for this case. Other tests are disabled for the similar case. That's why left it.

**Comment by arcuri82** [comment]
@seran 
another thing i realized now: when writing E2E using taint, you should add:
```
            setOption(args, "taintForceSelectionOfGenesWithSpecialization", "true")
            setOption(args, "discoveredInfoRewardedInFitness", "true")
```
those are experimental features that seem working, but not activated on by default yet

**Review by arcuri82 — APPROVED** [review]
